transport/node.go

```diff
@@ -0,0 +1,29 @@
+package transport
+
+import (
+	"go.uber.org/atomic"
+)
+
+type nodeStatus = *atomic.Bool
+
+const (
+	statusDown = false
+	statusUP   = true
+)
+
+type Node struct {
```

Comment:
@Michal-Leszczynski 

- The idiomatic go naming scheme for field `x` is `X` and `SetX` no need for Get
- IMO all fields here should NOT be exported but they should have getters, this is API for main package and it prevents from untrusted code to change it...
- Also SetStatus should not be exported, I'd consider removing it

Reply:
> For a DTO fields should be exported...
I got confused by your previous comment, but I agree with this approach.